daemon/logger/awslogs/cloudwatchlogs_test.go

```diff
@@ -198,15 +204,128 @@ func TestCreateAlreadyExists(t *testing.T) {
 	if err != nil {
 		t.Fatal("Expected nil err")
 	}
+	if !stream.open {
+		t.Errorf("Expected stream.open to be true, was false")
+	}
+}
+
+func TestCreateOpenAlreadySet(t *testing.T) {
+	mockClient := newClientSpy()
+	stream := &logStream{
+		client: mockClient,
+		open:   true,
+	}
+	err := stream.create()
+
+	if err != nil {
+		t.Fatal("Expected nil err")
+	}
+	if mockClient.createLogStreamCallCount != 0 {
+		t.Fatal("Did not expect createLogStreamCount to be called")
+	}
+}
+
+func TestLogClosed(t *testing.T) {
+	mockClient := newMockClient()
+	stream := &logStream{
+		client: mockClient,
+		open:   true,
+		closed: true,
+	}
+	err := stream.Log(&logger.Message{})
+	if err == nil {
+		t.Fatal("Expected non-nil error")
+	}
 }
 
+func TestLogBlocking(t *testing.T) {
+	mockClient := newMockClient()
+	stream := &logStream{
+		client:   mockClient,
+		open:     true,
+		messages: make(chan *logger.Message),
+	}
+
+	errorCh := make(chan error, 1)
+	done := make(chan bool)
+	go func() {
+		err := stream.Log(&logger.Message{})
+		errorCh <- err
+		done <- true
+	}()
+	select {
+	case <-errorCh:
+		t.Fatal("Expected stream.Log to block")
+	case <-time.After(1 * time.Second):
```

Comment:
Does this need to be a `time.After` instead of just a `default`? Seems like we are trying to mitigate race conditions, but sleeping like this is always racey.